app/src/main/java/org/dhis2/data/forms/dataentry/fields/scan/ScanBindings.kt

```diff
@@ -0,0 +1,73 @@
+package org.dhis2.data.forms.dataentry.fields.scan
+
+import android.view.View
+import android.view.inputmethod.EditorInfo
+import android.widget.ImageView
+import androidx.core.widget.doOnTextChanged
+import androidx.databinding.BindingAdapter
+import com.google.android.material.textfield.TextInputEditText
+import com.google.android.material.textfield.TextInputLayout
+import org.dhis2.Bindings.closeKeyboard
+import org.dhis2.R
+import org.hisp.dhis.android.core.common.ValueTypeRenderingType
+
+@BindingAdapter("action_icon")
+fun ImageView.setActionIcon(model: ScanTextViewModel) {
+    val iconRes = when (model.fieldRendering?.type()) {
+        ValueTypeRenderingType.QR_CODE -> R.drawable.ic_form_qr
+        ValueTypeRenderingType.BAR_CODE -> R.drawable.ic_form_barcode
+        else -> null
+    }
+
+    iconRes?.let { setImageResource(iconRes) }
+}
+
+@BindingAdapter(value = ["error_message", "warning_message"], requireAll = false)
+fun TextInputLayout.setErrorMessage(errorMessage: String?, warningMessage: String?) {
+    error = when {
+        errorMessage != null -> {
+            setErrorTextAppearance(R.style.error_appearance)
+            errorMessage
+        }
+        warningMessage != null -> {
+            setErrorTextAppearance(R.style.warning_appearance)
+            warningMessage
+        }
+        else -> null
+    }
+}
```

Comment:
is this being used in multiple times? or it will in the future? Do you think it makes sense to extract it to a common class?